sandy/core/cov.py

```diff
@@ -633,7 +633,7 @@ def log2norm_mean(self, mu):
         return np.log(mu_**2 / np.sqrt(np.diag(self.data) + mu_**2))
 
     def sampling(self, nsmp, seed=None, rows=None, pdf='normal',
-                 tolerance=None, relative=True, truncate=True):
+                 tolerance=None, relative=True, truncate=True, cutzeros=False):

```

Comment:
I am not even sure whether to keep it as a keyword argument.
I guess it should just be the default.
@GrimFe , your opinion?

Then, the `truncate` condition should not apply anymore to the uniform distribution, and since it did not apply either for the lognormal distribution, it only remains for the normal one.
For simplicity we might even remove it, since also for the normal distribution we always truncate .

Reply:
I see your point, but I like leaving the option to the user.
If the point is the number of keyword arguments, then one could think of using only the `truncate` condition, just differently applied to the gaussian distribution (symmetrical cut + setting of the tails to zero or to 2*mean), lognormal distribution (ignored) and uniform distribution (implementing the procedure now conditioned by `cutzeros`).
I agree that by default SANDY should not produce negative samples, but this is a matter of approach.

Reply:
Then, I guess we can converge on the following:

- all truncations should be applied by default and independently on the distribution
- there is no need for two kwargs `truncate` and `cutzeros`. If kept, they should be merged into one because they are basically doing the same
- keeping them (as @GrimFe prefers) or removing them (as I prefer) will be up to @EnricaBelfiore , making an assessment on the simplicity of consistently merging the two keywords into one, and on the maintainability of the method.

Reply:
The second option would require to remove the `relative` kwarg, due to the fact that negative samples are allowed if the sampling is performed with an absolute covariance matrix. If we are ok with removing that, I will remove these kwargs (`relative`, `cutzeros` and `truncate`) and the sampling method will only give positive/relative values

Reply:
Relative in theory is needed for sampling MF35 (covariance matrices are in absolute values).
Can't we just truncate between -1 and 1 (instead of 0 and 2) and then recenter into one if relative?

Reply:
In absolute terms I think we should cut between -mean and +mean (of the evaluation and not of the perturbation coefficient distribution) to avoid negative samples... so I would sample only positive values when the covariance is relative and allow negative samples (without symmetrical cut) when the covariance is absolute, removing the `truncate` and `cutzeros` kwargs